sources/packages/backend/apps/api/src/services/education-program-offering/custom-validators/has-valid-offering-period-for-funded-days.ts

```diff
@@ -20,32 +20,32 @@ class HasValidOfferingPeriodForFundedDaysConstraint
   implements ValidatorConstraintInterface
 {
   validate(studyBreaks: StudyBreak[], args: ValidationArguments): boolean {
-    const offeringMinDaysAllowedValue = this.getOfferingMinAllowedDays(args);
+    const offeringMinWeeksAllowedValue = this.getOfferingMinAllowedWeeks(args);
     const calculatedStudyBreaksAndWeeks = this.getCalculatedStudyBreaks(
       studyBreaks,
       args,
     );
     return (
-      calculatedStudyBreaksAndWeeks.fundedStudyPeriodDays >=
-        offeringMinDaysAllowedValue &&
+      calculatedStudyBreaksAndWeeks.totalFundedWeeks >=
+        offeringMinWeeksAllowedValue &&
       calculatedStudyBreaksAndWeeks.fundedStudyPeriodDays <=
         OFFERING_STUDY_PERIOD_MAX_DAYS
     );
   }
 
   defaultMessage(args: ValidationArguments) {
-    const offeringMinDaysAllowedValue = this.getOfferingMinAllowedDays(args);
-    return `The funded study amount of days is ineligible for StudentAid BC funding. Your dates must be between ${offeringMinDaysAllowedValue} to ${OFFERING_STUDY_PERIOD_MAX_DAYS} days.`;
+    const offeringMinWeeksAllowedValue = this.getOfferingMinAllowedWeeks(args);
+    return `The funded study amount of days is ineligible for StudentAid BC funding. Your offering must be at least ${offeringMinWeeksAllowedValue} weeks of study or longer to be eligible.`;
```

Comment:
The message isn't accurate with respect to logic anymore. 
The logic validates if the `offering funded weeks >= offeringMinWeeksAllowedValue` and also `offering funded days is <= 365 days`.  
Hence either the message must be updated to be in sync with the validation 
or
minimum funded weeks must use a separate constraint and message.

Reply:
@andrewsignori-aot @andrepestana-aot @guru-aot @bidyashish @katrina-aot-git 

Do you have any preference about what approach to take here - 

1. Same constraint and a common warning message

OR

2. New constraint and separate warning message. ?

Reply:
Option one would be more aligned with the current ticket ACs. The specific message to conciliate both into one sentence should come from business.

The second one can provide more precise warnings but it would also demand business providing separate content messages and a new banner to be displayed in the UI.

For the sake of this PR I would go with option 1.

Reply:
On second thought, the new messages are specific for the minimum validation, right?
If yes, should we not segregate the validations to achieve the ticket ACs?

Reply:
After the quick sync, the assumptions are below.
- Currently, we have 2 validations for the max 365.
  - The first validates the 365 between the start and end date.
  - The second validates the 365 funded days between the start and end date and also the minimum amount of funded days (to be changed to validate funded weeks).
- Both validators generate the same warning key that results in the same banner in the form.io UI and same message for the bulk upload.

### Proposed solution

- Keep the first decorator validating only the 365 between the start and end date (not the funded values).
- The first validator should display the banner where it is currently, right below the Start and End dates.
  - Business to provide the content message.
- The second validator will check only the minimum funded weeks, as requested in this ticket, and display the messages with the contents as requested in this ticket. The banner will be displayed right below the study breaks following the ticket initial requirements.